### PR TITLE
use configurable absolute url for sitemap locations in index file

### PR DESCRIPTION
**Pull request by s-diez**
See https://github.com/aimeos/ai-controller-jobs/issues/17

As this option is necessary and no default make sense, i added the same behavior as for location.

### CODE REVIEW

**Comment by aimeos** [comment]
The same applies to the product sitemap index file as well, doesn't it?
Can you please use `baseurl` instead of `base-url` to be in line with the other settings?

**Comment by s-diez** [comment]
> The same applies to the product sitemap index file as well, doesn't it?

Yes the PR contains both.

> Can you please use `baseurl` instead of `base-url` to be in line with the other settings?

Done

**Comment by aimeos** [comment]
Does it make sense to add a common sitemap baseurl setting so users don't need to set both? For example, in the TYPO3 scheduler task there should be a possibility to enter one URL and that value is required when saving the task.

File: controller/jobs/src/Controller/Jobs/Catalog/Export/Sitemap/Standard.php [hunk 1]
```diff
@@ -325,13 +329,46 @@ protected function createSitemapIndex( \Aimeos\MW\Container\Iface $container, ar
...
+		 * @see controller/jobs/catalog/export/sitemap/container/options
+		 * @see controller/jobs/catalog/export/sitemap/max-items
+		 * @see controller/jobs/catalog/export/sitemap/max-query
+		 * @see controller/jobs/catalog/export/sitemap/changefreq
+		 * @see controller/jobs/catalog/export/sitemap/location
+		 */
+		$urlConf = 'controller/jobs/catalog/export/sitemap/baseurl';
+
+		$baseUrl = $context->getConfig()->get($urlConf, '');
+		if( $baseUrl === '' )
```
Comment (aimeos) [inline controller/jobs/src/Controller/Jobs/Catalog/Export/Sitemap/Standard.php]: You can write that a bit shorter:
```
if( ( $baseUrl = $config->get( $urlConf ) ) == '' ) {
```

**Comment by s-diez** [comment]
Only one setting would be easier, but maybe not as flexible. The URL most likely corresponds to the "location" setting. I think we would then need to change this to a common setting too.

Regarding the scheduler task, the task handles many jobs so enforcing this setting when the user does not actually want a sitemap job is kinda weird. Again similar to the location setting which is required for the sitemap jobs but has to be set through the option "TypoScript Konfiguration".  Maybe it would be possible to add fields depending on the selected jobs or at least change the validation to required depending on jobs?

**Comment by aimeos** [comment]
You are right, the location needs to be set nevertheless

**Pull request merged**